src/modules/Invoice/Service.php

```diff
@@ -1205,81 +1207,38 @@ public function generatePDF($hash, $identity)
-                <p>Invoice date: " . $format->format(strtotime($invoice['created_at'])) . "</p>
-                <p>Due date: " . $format->format(strtotime($invoice['due_at'])) . "</p>
-                <p>Invoice status: " . ucfirst($invoice['status']) . "</p>
-            </div>
-
-            <h3 class='CompanyInfo'>Company</h3>
-            <div class='CompanyInfo'>" . $this->getSellerPdfHtml($invoice, $sellerLines) . "</div>
-
-            <h3 class='ClientInfo'>Client</h3>
-            <div class='ClientInfo'>" . $this->getBuyerPdfHtml($invoice, $buyerLines) . "</div>
-            <div class='Breakdown' style='top: " . (($buyerLines >= $sellerLines) ? (325 + (25 * $buyerLines)) : (325 + (25 * $sellerLines))) . "px'
-                <table style='width:100%'>
-                <tr>
-                    <th>#</th>
-                    <th>Product</th>
-                    <th>Quantity & Price</th>
-                    <th>Total</th>
-                </tr>
-        ";
-
-        $nr = 1;
-        foreach ($invoice['lines'] as $row) {
-            $html .=
-            "<tr>
-                <th>" . $nr++ . "</th>
-                <th>" . $row['title'] . "</th>
-                <th>" . $row['quantity'] . 'x ' . $this->money($row['price'], $currencyCode) . "</th>
-                <th>" . $this->money($row['total'], $currencyCode) . "</th>
-            </tr>";
-        }
-
-        $html .=
-            "<tr>
-            <th colspan='4'>___________________________________________________________________________________________</th>
-        </tr>";
-
-        if ($invoice['tax'] > 0) {
-            $html .=
-            "<tr>
-                <th class='right' colspan='3'>" . $invoice['taxname'] . " " . $invoice['taxrate'] . "% Tax:</th>
-                <th>" . $invoice['tax'] . $currencyCode . "</th>
-            </tr>";
-        }
-
-        if (isset($invoice['discount']) && $invoice['discount'] > 0) {
-            $html .=
-            "<tr>
-                <th class='right' colspan='3'>Discount:</th>
-                <th>" . $invoice['discount'] . $currencyCode . "</th>
-            </tr>";
-        }
-
-        $html .=
-        "<tr>
-            <th class='right' colspan='3'>Total:</th>
-            <th>" . $invoice['total'] . $currencyCode . "</th>
-        </tr>
-        </table>
-        </body>
-        </html>";
+        // new code
+        $vars = [
+            'currency_code' => $currencyCode,
+            'css'          => $CSS,
+            'html_logo' => $logoHtml,
+            'html_seller' => $this->getSellerPdfHtml($invoice, $sellerLines),
+            'html_seller_lines' => $sellerLines,
+            'html_buyer' => $this->getBuyerPdfHtml($invoice, $buyerLines),
+            'html_buyer_lines' => $buyerLines,
+            'invoice' => $invoice,
+        ];
+        $twigOptions = [
+            'paths' => [dirname(__DIR__) . DIRECTORY_SEPARATOR .  'Invoice' . DIRECTORY_SEPARATOR . 'pdf_template'],
```

Comment:
The `dirname` function isn't needed when paired with `__DIR__`. The old method used to be `dirname(__FILE__)`, but `__DIR__` gives the same result.